### DISCUSSION TITLE
Universal RSC Expo Router + Nativewind

### DISCUSSION THREAD

**Discussion by joernroeder** [nativewind/nativewind#1370] [Q&A]
I know this is as fresh as it can get, I'm playing with RSC by following their examples. https://expo.dev/blog/universal-react-server-components-developer-preview

following the usual install results in the following

```
Warning: ReactServerError: Cannot access Appearance.getColorScheme on the server. You cannot dot into a client module from a server component. You can only pass the imported name through.
```

**Comment by marklawlor** [comment]
Yep pretty fresh 😅 This is under active development, but will probably come after the reanimated 3.16 fixes

**Comment by EvanBacon** [comment]
React Native will need to be reworked to ensure there isn’t a top-level “use client” directive. Adding the top-level directive was hard and took a long time to get approval. The early developer preview is essentially for the express purpose of making these types of changes in packages—much easier to do with a working version of RSC.

**Comment by danstepanov** [comment]
gonna move this to a discussion as it is under active development

**Comment by tuboihirokidesu** [comment]
Hey @marklawlor  ! Just checking in to see if there’s a timeline for when these discussion points might be addressed. Thanks in advance for any updates!

**Comment by Yurii-Lutsyk** [comment]
Is there any progress on this problem?

**Comment by lukepighetti** [comment]
totally get that one of the reasons why this technical preview was launched was to have a working example to use in proposals for improving other areas of the ecosystem. that said, what's the current best strategy for shipping real-world apps with expo server components right now? we are getting this error and our project has Nativewind but we are just in the experimenting stage so we're not tied to any technical decision

**Reply by danstepanov** [reply]
I'm really not aware of one tbh. Also, I think the discussion around this question is better addressed to Expo rather than Nativewind folks.

**Comment by danstepanov** [comment]
Can y'all help me understand what the ask is of Nativewind here? Appreciate it, thanks.
